### PR TITLE
Add howto for zone-bundle tool

### PR DESCRIPTION
**Pull request by bnaecker**
Also fixes stupid directory iteration bug.

### CODE REVIEW

**Comment by bnaecker** [comment]
Friendly ping on this quick PR @jordanhendricks!

**Review by jordanhendricks — COMMENTED** [review]
Thanks for adding these! A few minor comments.

File: docs/zone-bundle.adoc [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+bundles. Full help is available with `zone-bundle --help`, but the most useful
+commands are:
+
+- `list-zones`: List the _current_ zones on the system.
+- `ls [filter]`: List all zone bundles that exist, possibly with a substring
+  filter. This includes bundles from zones that no longer exist.
+- `create`: Create a zone bundle immediately. They are also created
+  automatically when zones are destroyed.
+- `get`: Copy a zone-bundle from internal storage. By default, the file is
+  copied to the local directory with the same name as in internal storage, which
```
[the remark this answers is not in the document]
Comment (bnaecker) [reply docs/zone-bundle.adoc:26]: It is in debug datasets on each M.2. I think that's really only in the code itself. Should we add it here?

[the remark this answers is not in the document]
Comment (jordanhendricks) [reply docs/zone-bundle.adoc:26]: Yeah, that seems like a worthwhile implementation detail to capture here, I think.

**Comment by bnaecker** [comment]
Thanks for the notes @jordanhendricks! I think I've addressed them in 13f0d67d, but let me know if you have other suggestions.

File: docs/zone-bundle.adoc [hunk 2]
```diff
@@ -4,11 +4,11 @@
 
 = Zone bundle
 
-The `zone-bundle` tool is a binary that can be used to manage a tarball of
-debugging data collected from Oxide zones. The tool collects all log files on
-the system, including current, rotated and archived files. It also runs a series
-of debugging commands in the target zone, such as `netstat -an`. All of these
-items are packaged into a tarball that can be fetched from the system.
+`zone-bundle` tool can be used to manage a tarball of debugging data collected
```
Comment (jordanhendricks) [inline docs/zone-bundle.adoc:7]: nit: dropped `The` by accident?

**Review by jordanhendricks — APPROVED** [review]
one small nit, but looks good! Thanks for adding this.

**Pull request merged**